server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -94,6 +98,47 @@ function selectProtocolVersion(connectVersions: string[]): string {
     }
 }
 
+/**
+ * @returns true if throttled and sent nack; false if not throttled or no throttler provided.
+ */
+function checkThrottle(
+    throttler: core.IThrottler | undefined,
+    throttleId: string,
+    socket: core.IWebSocket,
+    logger: core.ILogger): boolean {
+    if (!throttler) {
+        return false;
+    }
+    const messageMetaData = {
+        key: throttleId,
+        weight: 1,
+        event_type: "throttling",
+    };
+    try {
+        throttler.incrementCount(throttleId);
+    } catch (e) {
+        if (e instanceof core.ThrottlingError) {
+            logger.info(`Throttled: ${throttleId}`, { messageMetaData: {
+                ...messageMetaData,
+                reason: e.message,
+                retryAfterInSeconds: e.retryAfter,
+            } });
+            const nackMessage = createNackMessage(
+                e.code,
+                NackErrorType.ThrottlingError,
+                e.message,
+                e.retryAfter);
```

Comment:
log before editing nack.

Reply:
Do you mean something other than the log on lines 121-125?

Reply:
Oops. I missed that. Sorry.